### DISCUSSION TITLE
Detecting when the cache is dirty

### DISCUSSION THREAD

**Discussion by apGribble** [cpm-cmake/CPM.cmake#273] [Q&A]
When using the CPM_SOURCE_CACHE option is there a way to raise a warning or error during the configure process if the cache has been modified since being checked out.

Intellisense makes it particularly easy to inadvertently end up modifying one of the source files in the cache that will effect all builds on that machine that use the associated library.

**Comment by TheLartians** [comment]
Hey @apGribble, thanks for the question!
Currently there is no option for that, but it's a great idea and I can definitely see how it would help to avoid mistakes. Perhaps it would make sense to create a feature request issue for us to have it on the roadmap.

**Reply by apGribble** [reply]
I've created a new issue #274 , thanks for your feedback
